dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/visitor/JspVisitor.java

```diff
@@ -290,14 +290,48 @@ else if (jspElementType.equals(Constants.JSP_PARAM_TYPE)) {
                     visitJspParamEnd(jspElement);
                 }
                 else if (jspElementType.equals(Constants.JSP_PARAMS_TYPE)) {
-                    visitJspParamsStart(jspElement);
-                    processChildren(jspElement);
-                    visitJspParamsEnd(jspElement);
+                    if(PagesVersionHandler.isPages30OrLowerLoaded()){
+                        visitJspParamsStart(jspElement);
+                        processChildren(jspElement);
+                        visitJspParamsEnd(jspElement);
+                    } else {
+                        // Changed for Page 3.1. jsp:params is a no-op (as it must be a child of jsp:plugin)
+                        // JSP must still valid contents/syntax within (3.1 Spec, section 5.8 <jsp:params>)
+                        if(this.getClass().equals(com.ibm.ws.jsp.translator.visitor.validator.ValidateJspVisitor.class)){
+                            if(com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled()&&logger.isLoggable(Level.FINEST)){
+                                logger.logp(Level.FINEST, CLASS_NAME, "processJspElement","Processing Validation for the jsp:params element");
+                            }
+                            visitJspParamsStart(jspElement);
+                            processChildren(jspElement);
+                            visitJspParamsEnd(jspElement);    
+                        } else {
+                            if(com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled()&&logger.isLoggable(Level.FINEST)){
+                                logger.logp(Level.FINEST, CLASS_NAME, "processJspElement","Skipping the jsp:params element as it is a no operation for Pages 3.1+");
+                            }
+                        }
+                    }
```

Comment:
I could reduce the repetition by using if(PagesVersionHandler.isPages30OrLowerLoaded() || this.getClass().equals(com.ibm.ws.jsp.translator.visitor.validator.ValidateJspVisitor.class )

but I think readability is clearer as it's is now  --  If (3.0 or or lower  ) else